docs/apis/core/dml/delegated-transactions.md

```diff
@@ -0,0 +1,80 @@
+---
+title: Delegated database transactions
+tags:
+  - core_dml
+  - DML
+  - core
+  - API
+---
+
+## General principles
+
+1. No matter of the nested nature, they are really **"delegated transactions"** (so outer levels get the control over inner ones), so we are going to apply that naming to enforce the real concept
+2. Code will **never** rely on rollback happening. It is only a measure to reduce (not to eliminate) DB garbled information
+3. Any problem related with the use of transactions (unfinished ones, unbalanced, finished twice...) will end **always** with one transaction_exception that will **always** end performing one DB rollback
+4. If one transaction (at any level) has been marked for rollback() there won't be any method to change it. Finally Moodle will perform the DB rollback
+5. If one transaction (at any level) has been marked for commit() it will be possible to change that status to rollback() in any outer level
+6. It will be **optional** to catch exceptions when using transactions, but if they are caught, then it is mandatory to mark the transaction for rollback()
+7. Any explicit rollback() call will pass the exception originating it, i.e. rollback($exception) to be re-thrown
+8. API will work in a natural and similar fashion to current recordsets one
+9. We will start **recommending** to use InnoDB for 2.0 in environmental tests. Possibly, at some point in the future that will become a **must**. But **it is not** for now (be warned!)
```

Comment:
Most of these could do with a little re-wording to clarify their meaning.

We've also been generally trying to avoid the use of `e.g.` and `i.e.` preferring "for example" and "that is" a lot of the time - only because most style guides (Microsoft, MDN, and others) suggest that it isn't as clear for non-native speakers.

Reply:
Notes from the MS style guide:
- i.e. - https://docs.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/i/ie
- e.g. - https://docs.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/e/eg

MDN guidance (see note below table): https://developer.mozilla.org/en-US/docs/MDN/Guidelines/Writing_style_guide#latin_abbreviations

Reply:
I have gone an reworded these a bit. It probably needs another review. 
A few comments about this section:
- Point 2. It seems that the author thought this to be very important (it's mentioned in two different areas of the document) but I don't find the comment useful at all.
- Point 8 does not seem useful at all.
- Point 9 I'm not sure what the status of testing is using transactions. Perhaps you can give some guidance here on what this means?